Lib/test/test_imaplib.py

```diff
@@ -70,7 +70,22 @@ def test_that_Time2Internaldate_returns_a_result(self):
         for t in self.timevalues():
             imaplib.Time2Internaldate(t)
 
+    @staticmethod
+    def _check_port(address='localhost', port=imaplib.IMAP4_PORT):
+        import socket
```

Comment:
I suggest to move the import at top level.

Reply:
What about no import whatsoever? ;)